src/h3lib/lib/algos.c

```diff
@@ -162,6 +162,17 @@ H3Error H3_EXPORT(maxGridDiskSize)(int k, int64_t *out) {
     if (k < 0) {
         return E_DOMAIN;
     }
+    if (k >= 13780510) {
```

Comment:
Nit: Should we assign this to a const?

Reply:
And/or a collection of constants for each resolution? I assume a much lower `k` at res 0 will run into similar problems?

Reply:
As this function does not accept the resolution we have no way of knowing if the `k` value will be larger than the grid. This would require an API change.

`k` that generates >122 cells at resolution 0 is also not the immediate issue. The issue is `k` values like INT32_MAX which would result in signed integer overflow, which is undefined behavior.

Reply:
We can do that without breaking backwards compatibility by creating a new function that does take the resolution, and recommending it over the old one? That would help prevent memory over-allocation for the smaller resolutions, and we could mark the current one as deprecated (but still with this fix).

I do agree with @nrabinowitz in any case that this would be better as a self-descriptive constant variable name.

Reply:
Added a comment for the constant. I'd defer a new function which limits the size to a future PR.